client/consul.go

```diff
@@ -100,15 +124,73 @@ func (c *ConsulClient) GetLicense(ctx context.Context, q *consulapi.QueryOptions
 	var license string
 
 	f := func(context.Context) error {
+		var err error
 		license, err = c.Operator().LicenseGetSigned(q)
+
+		// Process the error by wrapping it in the correct error types
 		if err != nil {
-			license = ""
-			return err
+			statusCode := getStatusCodeFromError(ctx, err)
+
+			// If we get a StatusNotFound assume that this is because CTS
+			// is connected to OSS Consul where this endpoint isn't available
+			// wrap in the appropriate error
+			if statusCode == http.StatusNotFound {
+				err = &NonEnterpriseConsulError{Err: err}
+			}
+
+			// non-retryable errors allows for termination of retries
+			if isStatusCodeRetryable(statusCode) {
+				return err
+			} else {
+				return &retry.NonRetryableError{Err: err}
+			}
 		}
 		return nil
 	}
 
+	var nonEnterpriseConsulError *NonEnterpriseConsulError
 	err = c.retry.Do(ctx, f, desc)
+	if err != nil {
+		if errors.As(err, &nonEnterpriseConsulError) {
+			c.logger.Warn("Unable to get license, this is most likely caused by CTS connecting to OSS Consul")
+		}
+	}
 
 	return license, err
 }
+
+func getStatusCodeFromError(ctx context.Context, err error) int {
+	// Extract the unexpected responseCode substring
+	re := regexp.MustCompile("Unexpected response code: ([0-9]{3})")
```

Comment:
For the MustCompile conversation, I was thinking it should be used since i'm providing the string literal statically. Therefore, if it failed, that would indicate a programmer error, so should panic?

Reply:
Talked about with Lorna, will make this a global